cypress/integration/projectSettings/project_settings.ts

```diff
@@ -290,628 +308,589 @@ describe("Repo Settings", { testIsolation: false }, () => {
+    });
 
-      it("Shows a navigation warning modal that lists the general page when navigating away from project settings", () => {
-        cy.contains("My Patches").click();
-        cy.dataCy("navigation-warning-modal").should("be.visible");
-        cy.dataCy("unsaved-pages").within(() => {
-          cy.get("li").should("have.length", 1);
-        });
-        cy.get("body").type("{esc}");
-      });
+    it("Does not default to repo value for display name", () => {
+      cy.dataCy("display-name-input").should("not.have.attr", "placeholder");
+    });
 
-      it("Shows the repo value for Batch Time", () => {
-        cy.dataCy("batch-time-input").should("have.attr", "placeholder");
+    it("Shows a navigation warning modal that lists the general page when navigating away from project settings", () => {
+      cy.dataCy("spawn-host-input").type("/test");
+      saveButtonEnabled();
+      cy.contains("My Patches").click();
+      cy.dataCy("navigation-warning-modal").should("be.visible");
+      cy.dataCy("unsaved-pages").within(() => {
+        cy.get("li").should("have.length", 1);
       });
+      cy.get("body").type("{esc}");
+    });
 
-      it("Clicking on save button should show a success toast", () => {
-        clickSave();
-        cy.validateToast("success", "Successfully updated project");
-      });
+    it("Shows the repo value for Batch Time", () => {
+      cy.dataCy("batch-time-input").should("have.attr", "placeholder");
+    });
 
-      it("Saves when batch time is updated", () => {
-        cy.dataCy("batch-time-input").type("12");
-        clickSave();
-        cy.validateToast("success", "Successfully updated project");
-      });
+    it("Clicking on save button should show a success toast", () => {
+      cy.dataCy("spawn-host-input").type("/test");
+      clickSave();
+      cy.validateToast("success", "Successfully updated project");
     });
 
-    describe("Variables page", () => {
-      before(() => {
-        cy.dataCy("navitem-variables").click();
-      });
+    it("Saves when batch time is updated", () => {
+      cy.dataCy("batch-time-input").type("12");
+      clickSave();
+      cy.validateToast("success", "Successfully updated project");
+    });
+  });
 
-      it("Successfully saves variables", () => {
-        cy.dataCy("add-button").click();
-        cy.dataCy("var-name-input").type("a");
-        cy.dataCy("var-value-input").type("1");
-        cy.dataCy("var-private-input").check({ force: true });
+  describe("Variables page", () => {
+    beforeEach(() => {
+      cy.dataCy("navitem-variables").click();
+      saveButtonEnabled(false);
+    });
 
-        cy.dataCy("add-button").click();
-        cy.dataCy("var-name-input").first().type("b");
-        cy.dataCy("var-value-input").first().type("2");
+    it("Successfully saves variables and then promotes them using the promote variables modal", () => {
+      // eslint-disable-next-line cypress/no-unnecessary-waiting
+      cy.wait(100);
```

Comment:
Can we assert on some variable or element changing? This tends to lead to flakiness if an action takes longer than this timeout.